webapp/channels/src/components/admin_console/access_control/table_editor/table_editor.tsx

```diff
@@ -0,0 +1,254 @@
+        }
+    };
+
+    if (!isTableMode) {
+        return (
+            <div className='table-editor__disabled-message'>
+                <i className='icon icon-alert-outline'/>
+                <FormattedMessage
+                    id='admin.access_control.table_editor.complex_expression'
+                    defaultMessage='Complex expression detected. Simple expressions editor is not available at the moment.'
+                />
+            </div>
+        );
+    }
+
+    return (
+        <div className='table-editor'>
+            <div className='table-editor__conditions'>
+                {rows.map((row, index) => (
+                    <div
+                        key={index}
+                        className='table-editor__condition'
+                    >
+                        <div className='table-editor__condition-attribute'>
+                            {row.attribute}
+                        </div>
+                        <div className='table-editor__condition-equals'>{'='}</div>
+                        <div className='table-editor__condition-value'>
+                            {row.value}
+                            <button
+                                className='table-editor__condition-remove'
+                                onClick={() => removeCondition(index)}
+                                disabled={disabled}
+                            >
+                                <i className='icon icon-close'/>
+                            </button>
+                        </div>
+                    </div>
+                ))}
+                {!disabled && getAvailableAttributes().length > 0 && (
+                    <div className='table-editor__add-condition'>
+                        {showAttributeDropdown ? (
+                            <div className='table-editor__dropdown-container'>
+                                <div className='table-editor__dropdown'>
+                                    {getAvailableAttributes().map((attr) => (
+                                        <div
+                                            key={attr.attribute}
+                                            className='table-editor__dropdown-section'
+                                        >
+                                            <div className='table-editor__dropdown-attribute'>
+                                                {attr.attribute}
+                                            </div>
+                                            <div className='table-editor__dropdown-values'>
+                                                {attr.values.map((value) => (
+                                                    <button
+                                                        key={value}
+                                                        className='table-editor__dropdown-value'
+                                                        onClick={() => addCondition(attr.attribute, value)}
```

Comment:
This is creating a new function on every render cycle. Create a new function for handle this click event and wrap it in `useCallback`, so any other function it calls also needs to go under `useCallback`. Normally, if we create a function in a component and its not wrapped in useCallback, its either wrong or can be a const function and can be moved outside of the component to reduce the amount of client side work on each render cycle.